src/main/java/com/puppycrawl/tools/checkstyle/checks/AvoidEscapedUnicodeCharactersCheck.java

```diff
@@ -348,7 +350,13 @@ public void beginTree(DetailAST rootAST) {
 
     @Override
     public void visitToken(DetailAST ast) {
-        final String literal = ast.getText();
+        // pitest does not like the following assignment guarded by
+        // a 'if' statement that checks token type. Since traditional string
+        // literals do not have preceeding whitespace or newline,
+        // 'stripIndentAndInitialNewLineFromTextBlock' has no effect
+        // on them and does not affect the functionality of this check.
```

Comment:
I think we should move this text to PR description. This comment is hard to understand

Reply:
I put it where it is so that when someone reads the code later, and sees that we call `CheckUtil.stripIndentAndInitialNewLineFromTextBlock()` on all token text even if it is a `CHAR_LITERAL` or `STRING_LITERAL`, I want them to easily understand why this was done, since it seems a bit odd to use this method on text that is not from a text block...
I can change it if you disagree.

Reply:
I can do `// see https://github.com/checkstyle/checkstyle/pull/9135 for explanation` and put this text in the PR description.

Reply:
@romani I put a detailed comment in the PR description, and just did a 
`// See https://github.com/checkstyle/checkstyle/pull/9135#issue-545666710` comment before the assignment.